src/controllers/organizationController.ts

```diff
@@ -317,204 +244,339 @@ export default class OrganizationController {
+                  "No request ids provided"
+                )
+            )
+          )
 
-  private async deleteAssociatedPreDraftOrganization(
-    userEmail: string
-  ): Promise<Option<IResponseErrorInternal | IResponseErrorConflict>> {
-    const errorOrMaybeOrganizationInstance = await getOrganizationInstanceFromDelegateEmail(
-      userEmail
-    );
-    if (isLeft(errorOrMaybeOrganizationInstance)) {
-      log.error(
-        "An error occurred reading data from db. %s",
-        errorOrMaybeOrganizationInstance.value
-      );
-      return some(
-        ResponseErrorInternal("An error occurred reading data from DB")
-      );
-    }
-    const organizationInstance = errorOrMaybeOrganizationInstance.value.toNullable();
-    if (organizationInstance) {
-      if (
-        organizationInstance.registrationStatus !==
-        OrganizationRegistrationStatusEnum.PRE_DRAFT
-      ) {
-        return some(
-          ResponseErrorConflict(
-            "There is already a registered organization associated to your account"
+          .map(requestIdCollection => ({
+            requestIds: requestIdCollection.items,
+            user: taskResults.user
+          }))
+      )
+      .chain<Pick<ITaskResults, "requestModels">>(taskResults =>
+        array
+          .traverse(taskEither)([...taskResults.requestIds], requestId =>
+            tryCatch<IResponseError, RequestModel>(
+              () =>
+                RequestModel.scope(RequestScope.INCLUDE_REQUESTER).findOne({
+                  where: { id: requestId }
+                }),
+              error =>
+                internalUnknownErrorHandler(
+                  error,
+                  "An error occurred while reading from the database."
+                )
+            )
+              .chain(
+                fromPredicate(
+                  _ => _ !== null,
+                  () =>
+                    ResponseErrorNotFound(
+                      "Request not found",
+                      `The request ${requestId} does not exist`
+                    )
+                )
+              )
+              .chain(
+                fromPredicate(
+                  _ =>
+                    _.type === RequestType.USER_DELEGATION ||
+                    _.type === RequestType.ORGANIZATION_REGISTRATION,
+                  () =>
+                    ResponseErrorValidation(
+                      "Bad request",
+                      `The type of request ${requestId} is invalid`
+                    )
+                )
+              )
+              .chain(
+                fromPredicate(
+                  _ => _.requester !== null,
+                  () => ResponseErrorInternal("Invalid internal data")
+                )
+              )
+              .chain(
+                fromPredicate(
+                  _ => _.requester!.email === taskResults.user.email,
+                  () => ResponseErrorForbiddenNotAuthorized
+                )
```

Comment:
another way is to try to decode the requester object using a type ie. `Requester = t.interface(...)`
in this way you can avoid the not-null assertion ("!")